### PR TITLE
ref(auto-assignment): Invalidate assignee cache after project ownership changes

### LINKED ISSUE #48564: Deleted rule applied to issues from CODEOWNERS file
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

1. https://sentry.zendesk.com/agent/tickets/85162 

Have not been able to replicate. Currently brought up by a single user

The issue is a rule that was deleted is still being used by Sentry to assign issues.

### Expected Result

The correct rule is applied

### Actual Result

The deleted rule is used

### Product Area

Issues

### Link

_No response_

### DSN

_No response_

### Version

sentry.ruby version 5.4.2

### PR DESCRIPTION
**Pull request by leeandher**
I believe this will solve the caching issue since if the GroupOwner was created via ownership, it may no longer be valid.

Resolves https://github.com/getsentry/sentry/issues/48564

### CODE REVIEW

**Review by JoshFerge — CHANGES_REQUESTED** [review]

File: src/sentry/tasks/post_process.py [hunk 1]
```diff
@@ -290,13 +292,7 @@ def handle_owner_assignment(job):
...
                         ):
                             # see ProjectOwnership.get_issue_owners
-                            issue_owners: Sequence[
-                                tuple[
-                                    Rule,
-                                    Sequence[Team | RpcUser],
-                                    str,
-                                ]
-                            ] = []
+                            issue_owners: Sequence[tuple[Rule, Sequence[Team | RpcUser], str]] = []
```
[the remark this answers is not in the document]
Comment (leeandher) [reply src/sentry/tasks/post_process.py:295]: The above isn't a code change, I removed a comma which reformatted the line. The other change in the file was auto-fix.

File: src/sentry/models/projectownership.py [hunk 2]
```diff
@@ -376,7 +376,7 @@ def process_resource_change(instance, change, **kwargs):
     autoassignment_types = ProjectOwnership._get_autoassignment_types(instance)
     if len(autoassignment_types) > 0:
         GroupOwner.invalidate_autoassigned_owner_cache(instance.project_id, autoassignment_types)
-
+    GroupOwner.invalidate_assignee_exists_cache(instance.project.id)
```
Comment (leeandher) [inline src/sentry/models/projectownership.py:379]: This is the only change, and it this runs as a signal when ProjectOwnership updates/deletes.

Comment (leeandher) [reply src/sentry/models/projectownership.py:379]: I will look into adding a test though :+1:

Comment (JoshFerge) [reply src/sentry/models/projectownership.py:379]: got it, makes sense! thanks.

**Review by JoshFerge — APPROVED** [review]

**Comment by JoshFerge** [comment]
thanks for adding a test! 🎉

**Pull request merged**